paddle/fluid/memory/allocation/allocator_facade.cc

```diff
@@ -295,7 +295,7 @@ class AllocatorFacadePrivate {
 
 // Pimpl. Make interface clean.
 AllocatorFacade::AllocatorFacade() : m_(new AllocatorFacadePrivate()) {}
-AllocatorFacade::~AllocatorFacade() { delete m_; }
+AllocatorFacade::~AllocatorFacade() {}
```

Comment:
Please add some comments on the reason why we remove `delete m_`.

Reply:
done